extensions/tableplus/src/utils.tsx

```diff
@@ -21,12 +21,34 @@ export async function fetchDatabases() {
     const groupList = plist.parse(fs.readFileSync(groupLocations, "utf8")) as ReadonlyArray<plist.PlistObject>;
     const plistInformation = plist.parse(fs.readFileSync(plistVersionPath, "utf8")) as Readonly<plist.PlistObject>;
 
-    const groups = new Map<string, Group>(
-      groupList.map((group) => [
-        group.ID.toString(),
-        { id: group.ID.toString(), name: group.Name.toString(), connections: [] },
-      ]),
-    );
+    const groups = new Map<string, Group>();
+    const parentGroups = new Map<string, string>();
+
+    groupList.forEach((group) => {
+      const groupId = group.ID.toString();
+      const groupName = group.Name.toString();
+      const parentGroupId = group.GroupID?.toString() || "";
+
+      groups.set(groupId, {
+        id: groupId,
+        name: groupName,
+        connections: [],
+      });
+
+      if (parentGroupId) {
+        parentGroups.set(groupId, parentGroupId);
+      }
+    });
+
+    groups.forEach((group, groupId) => {
+      const parentGroupId = parentGroups.get(groupId);
+      if (parentGroupId) {
+        const parentGroup = groups.get(parentGroupId);
+        if (parentGroup) {
```

Comment:
```suggestion
    parentGroups.forEach((parentGroupId, groupId) => {
      const parentGroup = groups.get(parentGroupId);
      if (parentGroup) {
        const group = groups.get(groupId);
        if (group) {
```

We could perhaps iterate over parentGroups directly, reducing the need to call groups.forEach again. This would improve readability and potentially performance.

Could you check that I didn't break anything? 😊

Reply:
are you still around @MatteoGauthier

Reply:
yes ! sorry for the late response